test/unit/BPool.t.sol

```diff
@@ -881,29 +881,170 @@ contract BPool_Unit_Rebind is BasePoolTest {
 }
 
 contract BPool_Unit_Unbind is BasePoolTest {
-  function test_Revert_NotController() private view {}
+  using LibString for *;
 
-  function test_Revert_NotBound() private view {}
+  struct Unbind_FuzzScenario {
+    uint256 tokenIndex;
+    uint256 balance;
+    uint256 denorm;
+    uint256 previousTokensAmount;
+    uint256 totalWeight;
+    address[] previousTokens;
+  }
 
-  function test_Revert_Finalized() private view {}
+  function _setValues(Unbind_FuzzScenario memory _fuzz) internal {
+    // Create mocks
+    _mockTransfer(_fuzz.previousTokens[_fuzz.tokenIndex]);
 
-  function test_Revert_Reentrancy() private view {}
+    // Set tokens
+    _setRandomTokens(_fuzz.previousTokensAmount);
+
+    // Set denorm and balance
+    _setRecord(
+      _fuzz.previousTokens[_fuzz.tokenIndex],
+      BPool.Record({bound: true, index: _fuzz.tokenIndex, denorm: _fuzz.denorm, balance: _fuzz.balance})
+    );
 
-  function test_Set_TotalWeight() private view {}
+    // Set finalize
+    _setFinalize(false);
+    // Set totalWeight
+    _setTotalWeight(_fuzz.totalWeight);
+  }
 
-  function test_Set_TokenArray() private view {}
+  function _assumeHappyPath(Unbind_FuzzScenario memory _fuzz) internal {
+    vm.assume(_fuzz.balance >= MIN_BALANCE);
+    vm.assume(_fuzz.totalWeight >= MIN_WEIGHT);
+    vm.assume(_fuzz.totalWeight <= MAX_TOTAL_WEIGHT - MIN_WEIGHT);
+    _fuzz.previousTokensAmount = bound(_fuzz.previousTokensAmount, 1, MAX_BOUND_TOKENS); // The token to unbind will be included inside the array
+    _fuzz.tokenIndex = bound(_fuzz.tokenIndex, 0, _fuzz.previousTokensAmount - 1);
+    _fuzz.denorm = bound(_fuzz.denorm, MIN_WEIGHT, _fuzz.totalWeight);
+    _fuzz.previousTokens = new address[](_fuzz.previousTokensAmount);
+    for (uint256 i = 0; i < _fuzz.previousTokensAmount; i++) {
+      _fuzz.previousTokens[i] = makeAddr(i.toString());
+    }
+  }
 
-  function test_Set_Index() private view {}
+  modifier happyPath(Unbind_FuzzScenario memory _fuzz) {
+    _assumeHappyPath(_fuzz);
+    _setValues(_fuzz);
+    _;
+  }
 
-  function test_Unset_TokenArray() private view {}
+  function test_HappyPath(Unbind_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
+    bPool.unbind(_fuzz.previousTokens[_fuzz.tokenIndex]);
```

Comment:
I'd delete this test case as there are others that already use the `happyPath`, verifying that it works.